// adminjs/lib/frontend/interfaces/action/build-action-click-handler.js
/* eslint-disable no-restricted-globals */
/* eslint-disable no-undef */
/* eslint-disable no-alert */

import { actionHasComponent } from './action-has-component.js';
import { actionHref } from './action-href.js';
import { buildActionCallApiTrigger } from './build-action-api-call-trigger.js';
export const buildActionClickHandler = options => {
  const {
    action,
    params,
    actionResponseHandler,
    navigate,
    modalFunctions
  } = options;
  const {
    openModal
  } = modalFunctions;
  const handleActionClick = event => {
    event.preventDefault();
    event.stopPropagation();
    const href = actionHref(action, params);
    const callApi = buildActionCallApiTrigger({
      params,
      action,
      actionResponseHandler
    });
    if (action.guard && actionHasComponent(action)) {
      const modalData = {
        modalProps: {
          variant: 'danger',
          label: 'confirm',
          title: action.guard
        },
        type: 'confirm',
        resourceId: params.resourceId,
        confirmAction: callApi
      };
      openModal(modalData);
      return;
    }
    if (href) {
      navigate(href);
    }
  };
  return handleActionClick;
};
